packages/cardano-services/src/Program/utils.ts

```diff
@@ -55,6 +56,13 @@ export const createDnsResolver = (config: RetryBackoffConfig, logger: Logger) =>
 
 export type DnsResolver = ReturnType<typeof createDnsResolver>;
 
+const httpServerOptionsToPostgresSsl = (options?: HttpServerOptions): ClientConfig['ssl'] =>
+  options?.postgresSsl
+    ? options.postgresSslCaFile
+      ? { ca: options.postgresSslCaFile }
```

Comment:
This file needs to be read before passing to ssl library.

Reply:
See https://github.com/input-output-hk/cardano-js-sdk/commit/51bee6ea0038d9de0e448e684fa002d4502521a6#diff-5935fb07258956001ade74f0edc794b473e06615d67973cc4d416ec157357be3R70-R71 for example

Reply:
In my quick refactor, I loaded it [here](https://github.com/input-output-hk/cardano-js-sdk/pull/354/files#diff-cd616b155b9729672e02921d8863a908cbaf8d92d935ca96f9708a083c87959bR92), but yes, doing it in the util will cover both entrypoints (we still have). @jvieiro please make this change, along with the test requested by @mkazlauskas below.